packages/application-config/src/formatters.ts

```diff
@@ -117,7 +118,58 @@ function entryPointUriPathToPermissionKeys<PermissionName extends string>(
   );
 }
 
+type TResourceAccessPermissionName<PermissionName extends string = ''> =
+  | `view${Capitalize<PermissionName>}`
+  | `manage${Capitalize<PermissionName>}`;
+
+const getPermissions = (
+  appConfig: JSONSchemaForCustomApplicationConfigurationFiles
+) => {
+  const additionalResourceAccessKeyToOauthScopeMap = (
+    appConfig.additionalOAuthScopes || []
+  ).reduce((previousOauthScope, { name, view, manage }) => {
+    const formattedResourceKey =
+      formatEntryPointUriPathToResourceAccessKey(name);
+    return {
+      ...previousOauthScope,
+      [`view${formattedResourceKey}`]: view,
+      [`manage${formattedResourceKey}`]: manage,
+    };
+  }, {} as Record<TResourceAccessPermissionName<PermissionName>, string[]>);
```

Comment:
This `PermissionName` type doesn't seem to be defined anywhere. In fact, it seems it's coming from the (global) DOM types 🤔

<img width="1066" alt="image" src="https://user-images.githubusercontent.com/1110551/192381331-f9d44087-d97f-4c9b-8dca-10e07afdce54.png">

Actually I also don't think we need the `TResourceAccessPermissionName` type and we can keep the record simple, otherwise we need the function overloading and all the related shenanigans.

```suggestion
  }, {} as Record<string, string[]>);
```

We can then cast the return type of `entryPointUriPathToResourceAccesses` to `Record<string, string>`.

Reply:
Updated [fbd2537](https://github.com/commercetools/merchant-center-application-kit/pull/2822/commits/fbd25376f21bfc7d81da512270a7e3dded6ea308)